cogs/roles.py

```diff
@@ -2,227 +2,121 @@
+    async def on_button_clicked(self, button: disnake.ui.Button, interaction: disnake.MessageInteraction, value=None):
+        """
+        Add or Remove Roles, when Button is clicked. Role gets added, if the user clicking the button doesn't have
+        the role already assigned, and removed, if the role is already assigned
+        """
 
-        emojis = {}
-        color_assignable = self.assignable_roles[1]
+        guild_roles = {str(role.id): role for role in interaction.guild.roles}
+        role = guild_roles.get(value)
 
-        # start with getting all emojis that are used in those roles as a dict
-        for emoji in self.bot.emojis:
-            if emoji.name in color_assignable:
-                emojis[emoji.name] = emoji
-
-        return emojis
-
-    def get_special_emojis(self):
-        """ Creates a dict for special role emojis """
-
-        return self.assignable_roles[2]
-
-    def get_key(self, role):
-        """ Get the key for a given role. This role is used for adding or removing a role from a user. """
-
-        for key, role_name in self.assignable_roles[0].items():
-            if role_name == role.name:
-                return key
-
-    @help(
-        category="info",
-        brief="Gibt die Mitgliederstatistik aus."
-    )
-    @commands.command(name="stats")
-    async def cmd_stats(self, ctx):
-        """ Sends stats in Chat. """
-
-        guild = ctx.guild
+        if role in interaction.author.roles:
+            await interaction.author.remove_roles(role)
+            await utils.send_dm(interaction.author, f"Rolle \"{role.name}\" erfolgreich entfernt")
+        else:
+            await interaction.author.add_roles(role)
+            await utils.send_dm(interaction.author, f"Rolle \"{role.name}\" erfolgreich hinzugefügt")
+
+        await interaction.response.defer()
+
+    @commands.slash_command(name="update-roles", description="Update Self-Assignable Roles")
+    @commands.default_member_permissions(moderate_members=True)
+    async def cmd_update_roles(self, interaction: disnake.ApplicationCommandInteraction):
+        """ Update all role assignment messages in role assignment channel """
+        await interaction.response.defer()
+
+        channel = await interaction.guild.fetch_channel(self.channel_id)
+        await channel.purge()
+        for role_category, roles in self.assignable_roles.items():
+            prefix = f"assign_{role_category}"
+            fields = []
+            buttons = []
+            value = f""
+            guild_roles = {role.name: role for role in interaction.guild.roles}
+
+            for key, role in roles.get("roles").items():
+                role_emoji = role.get('emoji') if role.get(
+                    'emoji') in emoji.UNICODE_EMOJI_ALIAS_ENGLISH else f"<{role.get('emoji')}>"
+                value += f"{role_emoji} : {role.get('name')}\n"
+                buttons.append({"emoji": role_emoji, "custom_id": f"{prefix}_{key}",
```

Comment:
Kann man Buttons (pro User) färben? 
Dann könnte man damit direktes Feedback geben, welche Rolle gerade gewählt ist.

Reply:
Jein. Schau mal in der kommando-zentrale auf das GIF. Auf dem Wege könnte man sowas machen und für die Zukunft wäre das auch mein Ziel sowas zu nutzen, aber ich glaube dafür müssen wir dann noch ganz andere Konzepte umbasteln.